src/mode/modeHandler.ts

```diff
@@ -124,6 +121,41 @@ export class VimState {
   public currentFullAction: string[] = [];
 
   /**
+   * Getters and setters for changing global state
+   */
+  public get searchStatePrevious(): SearchState[]{
```

Comment:
Instead of adding a bunch of shims for global state, can we just move these directly onto the globals object and allow consumers to use these getters/setters directly?